src/Utils/get.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Type-safe access of deep property of an object
+ *
+ * Thanks! https://github.com/claudepache/es-optional-chaining/issues/2#issuecomment-318680932
+ *
+ * @param obj                     Object to get deep property
+ * @param unsafeDataOperation     Function that returns the deep property
+ * @param valueIfFailOrUndefined  Value to return in case if there is no such property
+ */
+export function get<O, T>(
+  obj: O,
+  unsafeDataOperation: (x: O) => T,
+  valueIfFailOrUndefined?: any
+): T {
```

Comment:
Seeing as the default value for the fallback is `undefined` this return type should reflect that `T | undefined`